superset/assets/src/chart/Chart.jsx

```diff
@@ -123,6 +136,17 @@ class Chart extends React.PureComponent {
     if (chartStatus === 'failed') {
       return this.renderStackTraceMessage();
     }
+    
+    this.showOverlay = formData.show_overlay;
+
+    if(formData.hasOwnProperty('extra_filters') && formData['extra_filters'].length > 0) {
+      formData.show_overlay = false;
+      this.showOverlay = false;
+    }
+    
+    if (this.showOverlay) {
+      return this.renderChartOverlay();
+    }
 
```

Comment:
can we avoid to create this.showOverlay variable and use it from formData.show_overlay

Reply:
as discussed, converting this to a local variable instead of taking on `this` context. Done